elastic/kibana
# [Security Solution] Expandably flyout - Enable expandable flyout for generic events

## Summary

This PR enables the overview tab and left section insights for a generic event. When user to go `host` or `user` page and expand details for an event, in addition to table and json tab, they now have access to:

- Overview tab on the right section, which provide description of the event kind or event category (detail logic linked in comment), key insights such as highlighted fields, entities, prevalence and visualization previews (if available)
- Expanded details that includes entities details and prevalence details

Many sections are shared by the alert details flyout, which we are hoping to provide a unified experience when user opens the details flyout. 

#### When overview and expanded sections are enabled ####
- Ideally `event.kind` and `event.category` should be ecs compliant, meaning the field values are of `allowed_values` within [ecs definition](https://www.elastic.co/guide/en/ecs/current/ecs-event.html).
- If the field is not ecs compliant, and it does not fit the criteria to generate an event renderer, the overview tab and expanded sections are hidden
 
#### Variations depending on event kind ####
There is a variation of the about section depending on `event.kind`:
- `event.kind == 'event'`
   - This is the most general and common event document, hence we provide details at the `event.category` level.  
   - The title is also dynamic based on the category type (i.e if `event.category` is process, the `process.name` is displayed) 
- `event.kind != 'event'`
   - These are events that not as common/general as `event` so we are providing description at the `event.kind` level
   - The title matches the `event.kind` field
   - `event.category` is included as a list of categories present for the document 
<img width="1006" alt="image" src="https://github.com/elastic/kibana/assets/18648970/bb540c62-4346-4dc6-8c11-3ad6cdd1e7c9">

#### How to test ####
- Enable feature flag `expandableEventFlyoutEnabled`
- Generate some event data (the resolver generate data script is sufficient to the test main logic, to get the event renderer to show up, see comment on feeding additional data), alternatively, auditbeat and filebeat also feed event data.
- Go to Explore -> Host -> Events table -> expand event details

### Checklist

Delete any items that are not applicable to this PR.
